src/components/IRPeaksPanel.tsx

```diff
@@ -1,31 +1,117 @@
-import { ValueRenderers } from '..';
+import {
+  ColumnDef,
+  useReactTable,
+  getCoreRowModel,
+  flexRender,
+  SortingState,
+  getSortedRowModel,
+} from '@tanstack/react-table';
+import { useState } from 'react';
 
-import { Table } from './Table';
 import { IRPeak } from './context/data/DataState';
 
+type IRPeakPanelPreferences = Record<
+  keyof IRPeak,
+  {
+    visible?: boolean;
+    format?: (val: number | string) => string | number;
+    label?: string;
+  }
+>;
 export interface IRPeaksPanelProps {
+  /**
+   * peaks to display in the panel
+   *
+   */
   peaks: IRPeak[];
+  /**
+   * The columns to display in the table.
+   * @default {}
+   */
+  preferences?: IRPeakPanelPreferences;
 }
-export function IRPeaksPanel({ peaks }: IRPeaksPanelProps) {
-  const Rows = peaks.map(
-    ({ wavenumber, transmittance, absorbance, kind }, i) => (
-      <Table.Row key={i}>
-        <ValueRenderers.Number value={wavenumber} />
-        <ValueRenderers.Number value={transmittance} />
-        <ValueRenderers.Number value={absorbance} />
-        <ValueRenderers.Text value={kind} />
-      </Table.Row>
-    ),
+
+export function IRPeaksPanel(props: IRPeaksPanelProps) {
+  const {
+    peaks,
+    preferences = {
+      wavenumber: {},
+      transmittance: {},
+      absorbance: {},
+      kind: {},
+    },
+  } = props;
+  const columns: ColumnDef<IRPeak>[] = Object.entries(preferences).map(
+    ([key, { label = key, format = (x: number) => x }]) => ({
+      header: label,
+      accessorKey: key,
+      cell: ({ getValue }) => format(getValue()),
+    }),
   );
+
+  const [sorting, setSorting] = useState<SortingState>([]);
+  const table = useReactTable({
+    data: peaks,
+    columns,
+    state: {
+      sorting,
+      columnVisibility: Object.entries(preferences).reduce(
```

Comment:
This code looks quite complex. I hope that if the preferences.columns is an array is could make it simpler.

Anyway if a `map` is not possible please don't use `reduce` and use normal loops in a method called `getColumnVisibility`

Reply:
fixed